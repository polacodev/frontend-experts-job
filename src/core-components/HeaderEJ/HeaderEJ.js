import React from 'react';
import { Header, Item, Input } from 'native-base';

import IconEJ from '../IconEJ/IconEJ';

import color from '../../config/color/color';

const HeaderEJ = ({ searchText, handlerSearch = () => ({}) }) => {
  return (
    <Header searchBar rounded style={{ backgroundColor: color.light_gray }}>
      <Item>
        <IconEJ iconName="search" color={color.dark_gray} />
        <Input
          placeholder="Search"
          value={searchText}
          onChangeText={(text) => handlerSearch(text)}
        />
        <IconEJ iconName="users" color={color.dark_gray} />
      </Item>
    </Header>
  );
};

export default HeaderEJ;
