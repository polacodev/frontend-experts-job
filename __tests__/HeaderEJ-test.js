import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';

import HeaderEJ from '../src/core-components/HeaderEJ/HeaderEJ';
jest.useFakeTimers();
describe('<HeaderEJ />', () => {
  const onEventMock = jest.fn();

  test('renders <HeaderEJ /> correctly', () => {
    const tree = renderer
      .create(<HeaderEJ searchText="Mark" handlerSearch={onEventMock} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call handlerSearch event', () => {
    const { getByPlaceholderText } = render(
      <HeaderEJ searchText="Test" handlerSearch={onEventMock} />,
    );
    fireEvent(getByPlaceholderText('Search'), 'onChangeText', 'ab');
    expect(onEventMock).toHaveBeenCalledWith('ab');
  });

  test('should return undefined due the empty props', () => {
    const { container } = render(<HeaderEJ />);
    expect(container.firstChild).toBeUndefined();
  });
});
