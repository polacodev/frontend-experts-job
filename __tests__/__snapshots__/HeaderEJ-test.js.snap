// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderEJ /> renders <HeaderEJ /> correctly 1`] = `
<View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#e0e0e0",
      }
    }
  >
    <View
      rounded={true}
      searchBar={true}
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          Object {
            "backgroundColor": "#e0e0e0",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#CECDD2",
            "borderBottomWidth": 1,
            "borderColor": "transparent",
            "borderLeftWidth": 0,
            "borderRadius": 25,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "borderWidth": undefined,
            "flex": 1,
            "flexDirection": "row",
            "height": 30,
            "justifyContent": "flex-start",
            "marginLeft": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "color": "#414142",
                  "fontSize": 20,
                  "marginTop": 2,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                Object {
                  "color": "#9e9e9e",
                  "fontSize": 20,
                },
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="FontAwesome"
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          placeholder="Search"
          placeholderTextColor="#575757"
          rejectResponderTermination={true}
          style={
            Object {
              "alignSelf": "center",
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 30,
              "lineHeight": null,
              "paddingLeft": 5,
              "paddingRight": 5,
              "top": 1.5,
            }
          }
          underlineColorAndroid="rgba(0,0,0,0)"
          value="Mark"
        />
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "color": "#414142",
                  "fontSize": 20,
                  "marginTop": 2,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                Object {
                  "color": "#9e9e9e",
                  "fontSize": 20,
                },
              ],
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="FontAwesome"
        >
          
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
